src/main/java/io/wispforest/gadget/mixin/CustomPayloadC2SPacketMixin.java

```diff
@@ -1,27 +1,27 @@
 package io.wispforest.gadget.mixin;
 
 import net.minecraft.network.PacketByteBuf;
-import net.minecraft.network.packet.c2s.play.CustomPayloadC2SPacket;
-import org.spongepowered.asm.mixin.Final;
+import net.minecraft.network.packet.c2s.common.CustomPayloadC2SPacket;
 import org.spongepowered.asm.mixin.Mixin;
-import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 @Mixin(CustomPayloadC2SPacket.class)
 public class CustomPayloadC2SPacketMixin {
-    @Shadow @Final private PacketByteBuf data;
-
     private int gadget$stashedReaderIndex = 0;
 
-    @Inject(method = "write", at = @At("HEAD"))
+    @Inject(method = "write", at = @At(value = "INVOKE", 
+              target = "Lnet/minecraft/network/packet/CustomPayload;write(Lnet/minecraft/network/PacketByteBuf;)V", 
+              shift = At.Shift.BEFORE))
     private void stashReaderIndex(PacketByteBuf buf, CallbackInfo ci) {
-        gadget$stashedReaderIndex = data.readerIndex();
+        gadget$stashedReaderIndex = buf.readerIndex();
     }
 
-    @Inject(method = "write", at = @At("RETURN"))
+    @Inject(method = "write", at = @At(value = "INVOKE", 
+              target = "Lnet/minecraft/network/packet/CustomPayload;write(Lnet/minecraft/network/PacketByteBuf;)V", 
+              shift = At.Shift.AFTER))
     private void restoreReaderIndex(PacketByteBuf buf, CallbackInfo ci) {
-        data.readerIndex(gadget$stashedReaderIndex);
+        buf.readerIndex(gadget$stashedReaderIndex);
```

Comment:
The original reason I added this mixin was so that writing the custom payload packet wouldn't mutate the contained PacketByteBuf (so that I could write the packet twice), but it doesn't really make sense to rewind changes to the buffer you're writing to

Reply:
So then is this mixin just no longer needed since the payload can be written multiple times?

Reply:
https://github.com/wisp-forest/gadget/pull/33/commits/4fd7bfa57a995e57ee5c122d2982471179a9a184